test/parallel_api/dynamic_selection/sycl/test_auto_tune_policy_sycl.pass.cpp

```diff
@@ -336,27 +336,61 @@ int test_auto_submit_and_wait(UniverseContainer u, int best_resource) {
   return 0;
 }
 
-int main() { 
-  using policy_t = oneapi::dpl::experimental::auto_tune_policy<oneapi::dpl::experimental::sycl_backend>; 
+static inline void build_auto_tune_universe(std::vector<sycl::queue> &u) {
+  try {
+    auto device_cpu1 = sycl::device(sycl::cpu_selector_v);
+    sycl::queue cpu1_queue(device_cpu1);
+    run_sycl_sanity_test(cpu1_queue);
+    u.push_back(cpu1_queue);
+  } catch (sycl::exception) {
+    std::cout << "SKIPPED: Unable to run with cpu_selector\n";
+  }
+  try {
+    auto device_cpu2 = sycl::device(sycl::cpu_selector_v);
+    sycl::queue cpu2_queue(device_cpu2);
+    run_sycl_sanity_test(cpu2_queue);
+    u.push_back(cpu2_queue);
+  } catch (sycl::exception) {
+    std::cout << "SKIPPED: Unable to run with cpu_selector\n";
+  }
+  try {
+    auto device_cpu3 = sycl::device(sycl::cpu_selector_v);
+    sycl::queue cpu3_queue(device_cpu3);
+    run_sycl_sanity_test(cpu3_queue);
+    u.push_back(cpu3_queue);
+  } catch (sycl::exception) {
+    std::cout << "SKIPPED: Unable to run with cpu_selector\n";
+  }
+  try {
+    auto device_cpu4 = sycl::device(sycl::cpu_selector_v);
+    sycl::queue cpu4_queue(device_cpu4);
+    run_sycl_sanity_test(cpu4_queue);
+    u.push_back(cpu4_queue);
+  } catch (sycl::exception) {
+    std::cout << "SKIPPED: Unable to run with cpu_selector\n";
+  }
+}
+
+int main() {
+  using policy_t = oneapi::dpl::experimental::auto_tune_policy<oneapi::dpl::experimental::sycl_backend>;
+  std::vector<sycl::queue> u;
+  build_auto_tune_universe(u);
 
-  sycl::queue q1 = sycl::queue{sycl::cpu_selector_v};
-  sycl::queue q2 = sycl::queue{sycl::cpu_selector_v};
-  sycl::queue q3 = sycl::queue{sycl::cpu_selector_v};
-  sycl::queue q4 = sycl::queue{sycl::cpu_selector_v};
-  std::vector<sycl::queue> u = {q1, q2, q3, q4};
+  //If building the universe is not a success, return
+  if(u.size()==0) return 0;
```

Comment:
Adding some message that the universe size is 0 so aborting might be helpful.
Can be done later though.